packages/medusa/src/services/line-item.ts

```diff
@@ -484,23 +483,39 @@ class LineItemService extends TransactionBaseService {
     )
   }
 
+  async delete(ids: string[]): Promise<(LineItem | undefined | null)[]>
+  async delete(id: string): Promise<LineItem | undefined | null>
+
   /**
    * Deletes a line item.
    * @param id - the id of the line item to delete
    * @return the result of the delete operation
    */
-  async delete(id: string): Promise<LineItem | undefined | null> {
+  async delete<
+    T extends string | string[],
+    TResult = T extends []
+      ? (LineItem | undefined | null)[]
+      : LineItem | undefined | null
+  >(id: string | string[]): Promise<TResult> {
+    const ids = Array.isArray(id) ? id : [id]
+
     return await this.atomicPhase_(
       async (transactionManager: EntityManager) => {
         const lineItemRepository = transactionManager.withRepository(
           this.lineItemRepository_
         )
 
-        return await lineItemRepository
-          .findOne({ where: { id } })
-          .then(
-            async (lineItem) => lineItem && lineItemRepository.remove(lineItem)
-          )
+        const lineItems = await lineItemRepository.find({
+          where: { id: In(ids) },
+        })
+
+        let result
+        if (lineItems?.length) {
+          await lineItemRepository.remove(lineItems)
+          result = Array.isArray(id) ? lineItems : lineItems[0]
+        }
+
+        return result as TResult
```

Comment:
suggestion: return early for clarity (note: requires return type to be `Promise<TResult | undefined>` but that's not different from the original return type, also having `undefined[]` as a possible return seems a bit odd to me?)

```suggestion
        const lineItems = await lineItemRepository.find({
          where: { id: In(ids) },
        })

        
        if (!lineItems.length) {
          return
        }

        await lineItemRepository.remove(lineItems)
        return (Array.isArray(id) ? lineItems : lineItems[0]) as TResult
```

Reply:
excellent thanks :) plus some more changes ahah